Toolkit/ArcGISToolkit/JobManager.swift

```diff
@@ -167,49 +168,43 @@ public class JobManager: NSObject {
      Unregister a Job with the JobManager, using the Job's unique ID.
      Returns true if it found the Job and was able to unregister it.
      */
-    @discardableResult public func unregister(jobUniqueID: String) -> Bool{
+    @discardableResult public func unregister(jobUniqueID: String) -> Bool {
         let removed = keyedJobs.removeValue(forKey: jobUniqueID) != nil
         return removed
     }
     
     /// Clears the finished Jobs from the Job manager.
-    public func clearFinishedJobs(){
-        
+    public func clearFinishedJobs() {
         suppressSaveToUserDefaults = true
-        for (jobUniqueID, job) in keyedJobs{
-            if job.status == .failed || job.status == .succeeded{
-                keyedJobs.removeValue(forKey: jobUniqueID)
-            }
+        keyedJobs.filter({ $0.value.status == .failed || $0.value.status == .succeeded }).forEach {
```

Comment:
This potentially mutates `keyedJobs` multiple times. Instead, filter out the jobs you don't want and assign that back to `keyedJobs`:

```swift
public func clearFinishedJobs() {
    keyedJobs = keyedJobs.filter { (_, value) in
        let status = value.status
        return !(status == .failed || status == .succeeded)
    }
}
```

Reply:
Yeah. I had wondered about this, but since it has worked until now, I didn't want to change it. My assumption was that the enumerator is unaffected by the removals.

However, even if that's the case, this is also better since it will invoke writing to userDefaults less often (once for the entire collection of jobs, rather than once per change). I think.